core/unit_test/FloatingPointComparison.hpp

```diff
@@ -0,0 +1,322 @@
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOS_FLOATING_POINT_COMPARISON_HPP
+
+#include <Kokkos_Macros.hpp>
+#include <impl/Kokkos_Error.hpp>
+
+//____________________________________________________________________________//
+// FIXME remove definition below when available from Kokkos numeric traits
+//____________________________________________________________________________//
+#include <cfloat>
+namespace Kokkos {
+namespace Experimental {
+// clang-format off
+template <class> struct finite_min;
+template <> struct finite_min<float>       { static constexpr       float value = -FLT_MAX; };
+template <> struct finite_min<double>      { static constexpr      double value = -DBL_MAX; };
+template <> struct finite_min<long double> { static constexpr long double value = -LDBL_MAX; };
+template <class> struct finite_max;
+template <> struct finite_max<float>       { static constexpr       float value = FLT_MAX; };
+template <> struct finite_max<double>      { static constexpr      double value = DBL_MAX; };
+template <> struct finite_max<long double> { static constexpr long double value = LDBL_MAX; };
+// clang-format on
+}  // namespace Experimental
+}  // namespace Kokkos
+//____________________________________________________________________________//
+
+//____________________________________________________________________________//
+//____________________________________________________________________________//
+// Code below was taken from Boost.Test Version 1.74.0 and modified to include
+// __host__ __device__ annotations and remove Boost dependencies.
+//____________________________________________________________________________//
+//____________________________________________________________________________//
+
+// clang-format off
+
+//  (C) Copyright Gennadiy Rozental 2001.
+//  Distributed under the Boost Software License, Version 1.0.
+//  (See accompanying file LICENSE_1_0.txt or copy at
```

Comment:
Need to either include this license or remove the line

Reply:
Christian reimplemented the floating-point comparison and I dropped the file.  I rewrote history so it won't show up.